packages/server-core/src/user/strategies/apple.ts

```diff
@@ -118,7 +122,38 @@ export class AppleStrategy extends CustomOAuthStrategy {
     if (!existingEntity) {
       profile.userId = user.id
       const newIP = await super.createEntity(profile, params)
-      if (entity.type === 'guest') await this.app.service(identityProviderPath).remove(entity.id)
+      if (entity.type === 'guest' && profile.email) {
+        const profileEmail = profile.email
+        const existingIdentityProviders = await this.app.service(identityProviderPath).find({
+          query: {
+            $or: [
+              {
+                email: profileEmail
+              },
+              {
+                token: profileEmail
```

Comment:
same as above, double checking the token here is supposed to be the same email field

Reply:
Yep. In this particular strategy, if you're logging in with Apple SSO, we're checking if the email address associated with your Apple account matches the email associated with any other SSO accounts (identity-provider.email), or any magiclink login with that email (identity-provider.token). As mentioned before, email magiclink identity-providers store the email address on the `token` field.